### PR TITLE
refactor: invoke SelectERC20TokenDialog by message center

### PR DESCRIPTION
**Pull request by guanbinrui**
## Usage

```ts
const [, setOpenSelectERC20TokenDialog] = useRemoteControlledDialog(WalletMessages.events.selectERC20TokenDialogUpdated, useCallback((ev) => {
    if (ev.open || !ev.token) return
    // the user has already selected a token from the built-in token list.
}, []))

const onTokenChipClick = useCallback(() => {
    // the user clicked the token selector chip
}, [])
```

### CODE REVIEW

File: packages/maskbook/src/plugins/Wallet/messages.ts [hunk 1]
```diff
@@ -11,19 +13,19 @@ type SelectProviderDialogEvent =
...
       }
 
-type SelectWalletDialogEvent =
+export type SelectWalletDialogEvent =
     | {
           open: true
       }
     | {
           open: false
       }
```
Comment (septs) [inline packages/maskbook/src/plugins/Wallet/messages.ts:22]:
```suggestion
export type SelectWalletDialogEvent = { open: boolean }
```

**Comment by Tedko** [comment]
lgtm

**Review by septs — APPROVED** [review]

File: packages/maskbook/src/plugins/RedPacket/UI/RedPacketForm.tsx [hunk 2]
```diff
@@ -34,12 +34,12 @@ import ActionButton from '../../../extension/options-page/DashboardComponents/Ac
...
 import type { RedPacketJSONPayload } from '../types'
 import { resolveChainName } from '../../../web3/pipes'
-import { WalletMessages } from '../../Wallet/messages'
+import { SelectTokenDialogEvent, WalletMessages } from '../../Wallet/messages'
 import { useRemoteControlledDialog } from '../../../utils/hooks/useRemoteControlledDialog'
 import { useEtherTokenDetailed } from '../../../web3/hooks/useEtherTokenDetailed'
 import { useTokenBalance } from '../../../web3/hooks/useTokenBalance'
 import { EthereumMessages } from '../../Ethereum/messages'
-import { SelectERC20TokenDialog } from '../../Ethereum/UI/SelectERC20TokenDialog'
+import { SelectTokenDialog } from '../../Ethereum/UI/SelectTokenDialog'
```
Comment (zhouhanseng) [inline packages/maskbook/src/plugins/RedPacket/UI/RedPacketForm.tsx:42]: unused code

File: packages/maskbook/src/plugins/Trader/UI/trader/Trader.tsx [hunk 3]
```diff
@@ -26,10 +27,11 @@ import { useTradeStateComputed } from '../../trader/useTradeStateComputed'
 import { useTokenBalance } from '../../../../web3/hooks/useTokenBalance'
 import { getActivatedUI } from '../../../../social-network/ui'
 import { EthereumMessages } from '../../../Ethereum/messages'
-import { SelectERC20TokenDialog } from '../../../Ethereum/UI/SelectERC20TokenDialog'
+import { SelectTokenDialog } from '../../../Ethereum/UI/SelectTokenDialog'
```
Comment (zhouhanseng) [inline packages/maskbook/src/plugins/Trader/UI/trader/Trader.tsx:30]: unused code

**Pull request merged**